### PR TITLE
feat(cli): cli new ux

### PR DESCRIPTION
**Pull request by jayzhang**
feature link: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/24539544

1. new CLI engine
- improve "--help" UX
- move common logics into engine: telemetry, inputs mappings

2. fill in the gap between CLI and fx-core: add "npm run gen:cli" script in fx-core to automatically generator CLI command options from question model

Note: new UX will be enabled under env variable: TEAMSFX_CLI_NEW_UX=true, default is disable.

### CODE REVIEW

**Comment by wenytang-ms** [comment]
please run relative e2e or UI test case

**Comment by swatDong** [comment]
Are you going to rewrite all commands from `cmds/` to `commands/`? What will code owners be?

**Comment by jayzhang** [comment]
> Are you going to rewrite all commands from `cmds/` to `commands/`? What will code owners be?

The code owner is the feature owner of the corresponding component or API in FxCore.

**Comment by swatDong** [comment]
> > Are you going to rewrite all commands from `cmds/` to `commands/`? What will code owners be?
> 
> The code owner is the feature owner of the corresponding component or API in FxCore.

Then do you need code owners to review the new files under `commands/`?

File: packages/cli/src/commands/models/m365LaunchInfo.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {
+      command: "teamsfx m365 launchinfo --manifest-id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
+      description: "Get launch information of the acquired M365 App by Manifest ID",
+    },
+  ],
+  telemetry: {
+    event: TelemetryEvent.M365LaunchInfo,
+  },
+  handler: async (ctx) => {
+    logger.warning("This command is in preview.");
```
Comment (swatDong) [inline packages/cli/src/commands/models/m365LaunchInfo.ts:40]: The original one (https://github.com/OfficeDev/TeamsFx/blob/dev/packages/cli/src/cmds/m365/m365.ts#L80) sets log level to verbose to output detailed information.

Is it possible to ensure the same here?

Comment (jayzhang) [reply packages/cli/src/commands/models/m365LaunchInfo.ts:40]: We will unify the log verbose by "--verbose" option. And we will add a new verbose() API in logProvider interface. This involved defining standards on what content should be classified as verbose. I will drive for this discussion.

**Review by swatDong — DISMISSED** [review]

**Comment by jayzhang** [comment]
> > > Are you going to rewrite all commands from `cmds/` to `commands/`? What will code owners be?
> > 
> > 
> > The code owner is the feature owner of the corresponding component or API in FxCore.
> 
> Then do you need code owners to review the new files under `commands/`?

definitely

**Comment by jayzhang** [comment]
> please run relative e2e or UI test case

e2e test pass: https://github.com/OfficeDev/TeamsFx/actions/runs/5795180283/job/15707544958

**Review by sffamily — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Review by hund030 — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**